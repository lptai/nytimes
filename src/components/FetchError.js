import React from 'react';
import { Flex } from 'grid-styled';

const FetchError = ({ message, onRetry }) => (
  <Flex flexDirection={'column'}>
    <Flex justifyContent={'center'} mx={'auto'}>
      <p>Could not fetch articles. {message}</p>
    </Flex>
    <Flex justifyContent={'center'}>
      <button onClick={onRetry}>Retry</button>
    </Flex>
  </Flex>
);

export default FetchError;
