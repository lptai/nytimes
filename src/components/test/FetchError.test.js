import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import FetchError from './../FetchError';

test('Error message and retry button are shown', () => {
  const fetchError = shallow(<FetchError message={'Oops !!!'} />);
  const message = <p>Could not fetch articles. Oops !!!</p>;
  const retryBtn = <button>Retry</button>;

  expect(fetchError.contains(message)).toEqual(true);
  expect(fetchError.contains(retryBtn)).toEqual(true);
  expect(fetchError).toMatchSnapshot();
});
