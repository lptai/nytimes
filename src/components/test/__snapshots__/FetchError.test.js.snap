// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error message and retry button are shown 1`] = `
<Flex
  flexDirection="column"
  theme={
    Object {
      "breakpoints": Array [
        "40em",
        "52em",
        "64em",
      ],
      "fontSizes": Array [
        12,
        14,
        16,
        20,
        24,
        32,
        48,
        64,
        96,
        128,
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        32,
        64,
        128,
        256,
        512,
      ],
    }
  }
>
  <Flex
    justifyContent="center"
    mx="auto"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <p>
      Could not fetch articles. 
      Oops !!!
    </p>
  </Flex>
  <Flex
    justifyContent="center"
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <button>
      Retry
    </button>
  </Flex>
</Flex>
`;
